install/kubernetes/cilium/templates/cilium-operator-deployment.yaml

```diff
@@ -205,8 +205,13 @@ spec:
 {{- if and (or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (ge .Capabilities.KubeVersion.Minor "17") (gt .Capabilities.KubeVersion.Major "1")) .Values.enableCriticalPriorityClass }}
       priorityClassName: system-cluster-critical
 {{- end }}
+{{- if .Values.serviceAccounts.operator.create }}
       serviceAccount: cilium-operator
       serviceAccountName: cilium-operator
+{{- else }}
+      serviceAccount: {{ .Values.operator.serviceAccountName | quote }}
+      serviceAccountName: {{ .Values.operator.serviceAccountName | quote }}
+{{- end }}
```

Comment:
This seems a bit odd to me. Considering a situation where `serviceAccounts.operator.create: true` and `.Values.operator.serviceAccountName: cilium-temp-operator` the referenced service account name in the deployment will be incorrect as we use `.Values.operator.serviceAccountName` in operator rolebinding. 
We should be using the `.Values.operator.serviceAccountName` to reference the service account name irrespective of it being created or not. The service account will then be created only when `.Values.serviceAccounts.operator.create` is true.